File: test_data/csharp/test_main/aas_core_meta.v3rc2/input/snippets/Types/Extension/value_type_or_default.cs
/// <summary>
/// Return the <see cref="Aas.Extension.ValueType" /> or the default value
/// if it has not been set.
/// </summary>
public Aas.DataTypeDefXsd ValueTypeOrDefault()
{
    return (this.valueType != null)
        ? this.valueType
        : Aas.DataTypeDefXsd.String;
}

File: dev/test_data/main/golang/expected/aas_core_meta.v3/input/snippets/Types/Extension/value_type_or_default.go
// Return [IExtension.ValueType] or the default value
// if it has not been set.
func (_RECEIVER_ *_STRUCT_NAME_) ValueTypeOrDefault() DataTypeDefXSD {
	v := _RECEIVER_.ValueType()
	if v == nil {
		return DataTypeDefXSDString
	}

	return *v
}
